api/api/services/sendgrid_mail_global_send.py

```diff
@@ -0,0 +1,172 @@
+        "SENDGRID_SENDER", "YOUR_SENDGRID_SENDER")
+    SENDGRID_API_KEY = os.getenv(
+        "SENDGRID_API_KEY", "YOUR_SENDGRID_KEY")
+    TEMPLATE_ID_WELCOME = os.getenv(
+        "SENDGRID_TEMPLATE_ID_WELCOME", "YOUR_TEMPLATE_ID_WELCOME")
+    TEMPLATE_ID_INFORMATION = os.getenv(
+        "SENDGRID_TEMPLATE_ID_INFORMATION", "YOUR_TEMPLATE_ID_INFORMATION")
+    TEMPLATE_ID_PROMOTION = os.getenv(
+        "SENDGRID_TEMPLATE_ID_PROMOTION", "YOUR_TEMPLATE_ID_PROMOTION")
+    TEMPLATE_ID_SURVEY = os.getenv(
+        "SENDGRID_TEMPLATE_ID_SURVEY", "YOUR_TEMPLATE_ID_SURVEY")
+    TEMPLATE_ID_SURVEY_SCRIPTING = os.getenv(
+        "SENDGRID_TEMPLATE_ID_SURVEY_SCRIPTING", "YOUR_TEMPLATE_ID_SURVEY_SCRIPTING")
+
+
+class SendgridClient:
+    """Sendgrid integration API"""
+
+    def __init__(self, recipient: str, sender: Optional[str] = SendgridParameters.DEFAULT_SENDER):
+        self.logger = logging.getLogger(__name__)
+        logging.basicConfig(level=logging.INFO)
+
+        self.sendgrid_client = SendGridAPIClient(
+            api_key=SendgridParameters.SENDGRID_API_KEY)
+        self.mail = Mail(sender, To(recipient), subject=None)
+
+    def extract_first_name(self, email: str):
+        """Extract first name from email using regex"""
+        match = re.match(r"^[^._@]+", email)
+        return match.group() if match else ""
+
+    def set_template_data(self, data: dict, recipient_email: str, username: Optional[str] = None):
+        """Set dynamic data for the template"""
+        first_name = self.extract_first_name(recipient_email)
+        data["username"] = username or first_name
+        self.mail.dynamic_template_data = data
+
+    def send(self):
+        """Send email"""
+        if os.environ.get('TEST', '0') == '1':
+            self.logger.warning('[DEBUG] Email has hit .send() function')
+            return
+
+        try:
+            return self.sendgrid_client.send(self.mail)
+        except Exception as error:
+            self.logger.exception(error)
+            raise
+
+
+def get_recipients():
+    url = "https://v0ron.djnn.sh/api/mailing-list"
+    response = requests.get(url)
+    return response.json().get("results", [])
+
+
+def get_recipients_with_profile(token):
+    headers = {"Authorization": f"Token {token}"}
+
+    pentester_data = requests.get(
+        "https://v0ron.djnn.sh/api/pentester", headers=headers).json().get("results", [])
+    manager_data = requests.get(
+        "https://v0ron.djnn.sh/api/manager", headers=headers).json().get("results", [])
+
+    recipients_with_profile = {}
+    for data in pentester_data + manager_data:
+        email = data["auth"]["email"]
+        profile_image = data["auth"]["profile_image"]
+        recipients_with_profile[email] = profile_image
+
+    return recipients_with_profile
+
+
+if __name__ == "__main__":
+    token = '93043e1e16490d5acae5fa51110aee3a00e6fb92582cfb7b985c0fd8ffa913ce893c4bcc9815a45fdf30732514f9d83b0b0a32dca3554c3c706680ebb6d47a25'
```

Comment:
what is this ?

Reply:
I think you should remove this, there is no reason for this to be in our back-end. If you have test scripts or something, they should be in the `tests/` folder, not here.